chain/consensus/mir/manager.go

```diff
@@ -157,11 +157,14 @@ func NewManager(ctx context.Context,
 
 	params := trantor.DefaultParams(initialMembership)
 	params.Iss.SegmentLength = cfg.Consensus.SegmentLength // Segment length determining the checkpoint period.
-	params.Mempool.MaxTransactionsInBatch = 1024
+	if cfg.Consensus.MaxProposeDelay != 0 {
+		params.Iss.MaxProposeDelay = cfg.Consensus.MaxProposeDelay * time.Second
+	}
+	params.Iss.ConfigOffset = cfg.Consensus.ConfigOffset
 	params.Iss.AdjustSpeed(1 * time.Second)
-	params.Iss.ConfigOffset = ConfigOffset
 	params.Iss.PBFTViewChangeSNTimeout = 6 * time.Second
 	params.Iss.PBFTViewChangeSegmentTimeout = 6 * time.Second
```

Comment:
If `MaxProposeDelay` is configurable, these values must either be configurable as well, or must be derived from `MaxProposeDelay` and `SegmentLength`.
- `PBFTViewChangeSNTimeout` says how long the maximal time span is between two batches proposed by a single proposer can be before a PBFT view change is triggered. It must always be at least `MaxProposeDelay` plus a reserve of at least a few seconds, otherwise we'll be drowning in view changes.
- `PBFTViewChangeSegmentTimeout` is the same kind of timeout, but for segments (subsets of blocks). I.e., how long at most it can take a segment (consisting of `SegmentLength` blocks) to be finished. It has to be set at least to `MaxProposeDelay * SegmentLength` plus some reserve of at least a few seconds.

I'd probably go with having them configurable as well, while deriving the default values (if nothing is configured) from the existing parameters, something like `max(MaxProposeDelay + 5, 6)` and `max((MaxProposeDelay + 2) * SegmentLength + 3, 6)`, respectively.

Reply:
I would suggest deriving them for now, so we don't force the user to have to reason about this, WDYT?

Reply:
Added. @matejpavlovic Please review https://github.com/consensus-shipyard/lotus/pull/123/commits/fd81ccbc0b0afeb4c7d166249757d9d8c5d66142.